specs/GetFavicon.md

```diff
@@ -0,0 +1,155 @@
+# Background
+A favicon (favorite icon) is a tiny icon included along with a website, which is displayed in places like the browser's address bar, page tabs and bookmarks menu. Developers would like to have an API which allows them to retrieve the Favicon of a webpage, if it has been set, as well as get an update whenever the favicon has changed.
+
+# Description
+We propose a new Webview2 event which would allow developers to access the current Favicon of a page and be notified when the favicon changes. This means when a page first loads, it would raise the FaviconChanged event since before parsing the HTML document there is no known favicon. DOM or JavaScript may change the Favicon, causing the event to be raised again for the same document.
+
+# Examples
+## Win32 C++ Registering a listener for favicon changes
+```cpp
+    CHECK_FAILURE(m_webView->add_FaviconChanged(
+        Callback<ICoreWebView2FaviconChangedEventHandler>(
+            [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT {
+                
+                wil::unique_cotaskmem_string url;
+                CHECK_FAILURE(sender->get_FaviconUri(&url));
+                wil::com_ptr<IStream> pStream = SHCreateMemStream(nullptr, 0);
+
+                sender->GetFavicon(
+                    COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG, 
+                    pStream,
+                    Callback<
+                        ICoreWebView2GetFaviconCompletedHandler>(
+                        [&pStream, this](HRESULT code) -> HRESULT {
+                                    Gdiplus::Bitmap* pBitmap = new Gdiplus::Bitmap(pStream);
+                                    HICON icon;
+                                    pBitmap->GetHICON(&icon);
+                                    SendMessage(
+                                        m_appWindow->GetMainWindow(), WM_SETICON, ICON_SMALL,
+                                        (LPARAM)icon);
+                            return S_OK;
+                        })
+                        .Get());
+
+                return S_OK;
+            })
+            .Get(),
+        &m_faviconChangedToken));  
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    System.IO.Stream stream = new System.IO.MemoryStream();
+    await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png,
+        stream);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
+
+};
+```
+# API Notes
+If a Web page does not have a Favicon, then the FaviconChanged event
+is fired when the page is navigated to. The Favicon would be an
+empty image and empty Uri. The user is expected to handle this scenario.
+Otherwise we fire the FaviconChanged with an observed change to the
+Favicon. In that scenario, we would provide the Uri of the Favicon we
```

Comment:
I guess you mean the properties are updated with the new favicon value. Saying we provide them while raising the event makes it sound like they're on the event args.

Also instead of 'we' please use the type or object or whatever that is doing it.

```suggestion
Favicon. In that scenario, the CoreWebView2 has an updated value for the FaviconUri property, and the GetFavicon method to match the updated favicon.
```